### PR TITLE
fix: remove Cursor#transformStream

### PR DESCRIPTION
**Pull request by reggi**
NODE-2324

### CODE REVIEW

File: test/functional/cursor.test.js [hunk 1]
```diff
@@ -4093,7 +4093,7 @@ describe('Cursor', function () {
     });
   });
 
-  const testTransformStream = (config, _done) => {
+  const testStream = (config, _done) => {
```
Comment (emadum) [inline test/functional/cursor.test.js:4096]:
```suggestion
  const testTransformStream = (config, _done) => {
```

File: test/functional/cursor.test.js [hunk 2]
```diff
@@ -4152,15 +4152,15 @@ describe('Cursor', function () {
     const config = {
       client: client,
       configuration: configuration,
-      collectionName: 'transformStream-test-transform',
+      collectionName: '-test-transform',
```
Comment (emadum) [inline test/functional/cursor.test.js:4155]:
```suggestion
      collectionName: 'stream-test-transform',
```

File: test/functional/cursor.test.js [hunk 3]
```diff
@@ -4152,15 +4152,15 @@ describe('Cursor', function () {
...
       transformFunc: doc => ({ _id: doc._id, b: doc.a.b, c: doc.a.c }),
       expectedSet: new Set(expectedDocs)
     };
 
-    testTransformStream(config, done);
+    testStream(config, done);
```
Comment (emadum) [inline test/functional/cursor.test.js:4160]:
```suggestion
    testTransformStream(config, done);
```

File: test/functional/cursor.test.js [hunk 4]
```diff
@@ -4176,7 +4176,7 @@ describe('Cursor', function () {
...
```
Comment (emadum) [inline test/functional/cursor.test.js:4179]:
```suggestion
    testTransformStream(config, done);
```

**Review by emadum — APPROVED** [review]
LGTM 👍  just minor nits; technically we're still testing a transform stream since we're passing a transform function into `.stream`, so I suggested reverting the function name for clarity.

**Review by nbbeeken — APPROVED** [review]
I concur with Eric's suggestions, lgtm!

**Review by mbroadst — APPROVED** [review]
LGTM, but I think you should rename the commit `feat!: ....` so that we catch the breaking change in the release notes

**Pull request merged**